app/controllers/todos_controller.rb

```diff
@@ -12,7 +12,11 @@ def new
 
   def create
     @todo = current_user.todos.create(todo_params)
-    redirect_to todos_path, notice: 'Item was saved' 
+    if @todo.save 
```

Comment:
`@todo` is already saved to the database (if its valid) before here (line 14's `create` will write a valid Todo to the database). There's no need to call `save` and potentially incur the cost of another trip to the database here on line 15.

On line 15, consider using `if @todo.persisted?` instead of `@if @todo.save`.